### PR TITLE
[LoRA serialization] fix: duplicate unet prefix problem.

### PR DESCRIPTION
**Pull request by sayakpaul**
# What does this PR do?

Fixes: https://github.com/huggingface/diffusers/issues/5977

Trained a LoRA: https://huggingface.co/sayakpaul/corgy_dog_LoRA/tree/main (Colab: https://colab.research.google.com/gist/sayakpaul/bcd42382f111d5c0e3bf8b37b55342ff/sdxl_dreambooth_lora_.ipynb)

```python
from huggingface_hub import hf_hub_download
import safetensors.torch 

repo_id = "sayakpaul/corgy_dog_LoRA"
filename = "pytorch_lora_weights.safetensors"

path = hf_hub_download(repo_id=repo_id, filename=filename)
sd = safetensors.torch.load_file(path)
print(list(sd.keys())[:5])
```

Doing the above yields:

```bash
['unet.down_blocks.1.attentions.0.transformer_blocks.0.attn1.to_k.lora.down.weight',
 'unet.down_blocks.1.attentions.0.transformer_blocks.0.attn1.to_k.lora.up.weight',
 'unet.down_blocks.1.attentions.0.transformer_blocks.0.attn1.to_out.0.lora.down.weight',
 'unet.down_blocks.1.attentions.0.transformer_blocks.0.attn1.to_out.0.lora.up.weight',
 'unet.down_blocks.1.attentions.0.transformer_blocks.0.attn1.to_q.lora.down.weight']
```

Cc: @apolinario

### CODE REVIEW

File: src/diffusers/loaders/lora.py [hunk 1]
```diff
@@ -802,29 +807,21 @@ def save_lora_weights(
...
-                else text_encoder_lora_layers
-            )
+        if unet_lora_layers:
+            state_dict.update(pack_weights(unet_lora_layers, "unet"))
 
-            text_encoder_lora_state_dict = {
-                f"{cls.text_encoder_name}.{module_name}": param for module_name, param in weights.items()
-            }
-            state_dict.update(text_encoder_lora_state_dict)
+        if text_encoder_lora_layers:
```
Comment (sayakpaul) [inline src/diffusers/loaders/lora.py:823]: To keep it clean and unified with respect to how it's done for the SDXL LoRA writer.

File: src/diffusers/training_utils.py [hunk 2]
```diff
@@ -67,7 +67,7 @@ def unet_lora_state_dict(unet: UNet2DConditionModel) -> Dict[str, torch.Tensor]:
                 current_lora_layer_sd = lora_layer.state_dict()
                 for lora_layer_matrix_name, lora_param in current_lora_layer_sd.items():
                     # The matrix name can either be "down" or "up".
-                    lora_state_dict[f"unet.{name}.lora.{lora_layer_matrix_name}"] = lora_param
+                    lora_state_dict[f"{name}.lora.{lora_layer_matrix_name}"] = lora_param
```
Comment (sayakpaul) [inline src/diffusers/training_utils.py:70]: This was the main culprit!

**Comment by apolinario** [comment]
Do you think we could merge @patrickvonplaten ?

File: src/diffusers/loaders/lora.py [hunk 3]
```diff
@@ -391,6 +391,10 @@ def load_lora_into_unet(
         # their prefixes.
         keys = list(state_dict.keys())
 
+        if all(key.startswith("unet.unet") for key in keys):
+            deprecation_message = "Keys starting with 'unet.unet' are deprecated."
+            deprecate("unet.unet keys", "0.25", deprecation_message)
```
Comment (patrickvonplaten) [inline src/diffusers/loaders/lora.py:396]:
```suggestion
            deprecate("unet.unet keys", "0.27", deprecation_message)
```

We're already at 0.25

**Comment by patrickvonplaten** [comment]
Thanks a mille for fixing this @sayakpaul !

**Review by patrickvonplaten — APPROVED** [review]
Happy to merge once deprecation is updated to 0.27

**Pull request merged**